stl/inc/limits

```diff
@@ -35,12 +34,21 @@ _STL_DISABLE_CLANG_WARNINGS
 #undef new
 
 _STD_BEGIN
-_EXPORT_STD enum float_denorm_style { // constants for different IEEE float denormalization styles
+#ifdef _BUILD_STD_MODULE // TRANSITION, _Pragma("warning(disable : 4996)") doesn't work in modules

```

Comment:
We're able to suppress 4996 deprecations successfully elsewhere. What happens if you try moving the DISABLE/RESTORE macros to surround the structs, instead of the static data members? (As usual, please try this by locally running the test instead of pushing a commit, in case it doesn't work.)

Reply:
> What happens if you try moving the DISABLE/RESTORE macros to surround the structs, instead of the static data members?

I tried this way with 17.6 but it didn't work...

Reply:
Hmm, I can investigate when I have time. Thanks for trying!

Reply:
I have reported VSO-1752496 "Standard Library Modules: Suppressing C4996 deprecation warnings doesn't work with `static constexpr` data members".

<details>
<summary>Click to expand repro:</summary>

```
C:\Temp>type meow.ixx
```
```cpp
export module meow;

export enum [[deprecated]] FloatDenormStyle {
    denorm_indeterminate = -1,
    denorm_absent        = 0,
    denorm_present       = 1,
};

struct NumBase {
#pragma warning(push)
#pragma warning(disable : 4996)
    [[deprecated]] static constexpr FloatDenormStyle has_denorm = denorm_absent;
    [[deprecated]] static constexpr bool has_denorm_loss        = false;
#pragma warning(pop)
};

struct NumIntBase : NumBase {
    static constexpr bool is_integer = true;
};

struct NumFloatBase : NumBase {
    static constexpr bool is_integer = false;

#pragma warning(push)
#pragma warning(disable : 4996)
    [[deprecated]] static constexpr FloatDenormStyle has_denorm = denorm_present;
#pragma warning(pop)
};

export template <typename T>
struct NumericLimits : NumBase {};

template <>
struct NumericLimits<int> : NumIntBase {
    static constexpr int digits = 31;
};

template <>
struct NumericLimits<double> : NumFloatBase {
    static constexpr int digits = 53;
};
```
```
C:\Temp>type purr.cpp
```
```cpp
import meow;

static_assert(NumericLimits<int>::digits == 31);
static_assert(NumericLimits<int>::is_integer == true);

static_assert(NumericLimits<double>::digits == 53);
static_assert(NumericLimits<double>::is_integer == false);

#pragma warning(push)
#pragma warning(disable : 4996)
static_assert(NumericLimits<int>::has_denorm == denorm_absent);
static_assert(NumericLimits<int>::has_denorm_loss == false);

static_assert(NumericLimits<double>::has_denorm == denorm_present);
static_assert(NumericLimits<double>::has_denorm_loss == false);
#pragma warning(pop)

int main() {}
```
```
C:\Temp>cl /EHsc /nologo /W4 /std:c++latest /MT /Od /c meow.ixx
meow.ixx

C:\Temp>cl /EHsc /nologo /W4 /std:c++latest /MT /Od purr.cpp meow.obj
purr.cpp
C:\Temp\meow.ixx(12): warning C4996: 'FloatDenormStyle': was declared deprecated
C:\Temp\meow.ixx(3): note: see declaration of 'FloatDenormStyle'
C:\Temp\meow.ixx(26): warning C4996: 'FloatDenormStyle': was declared deprecated
C:\Temp\meow.ixx(3): note: see declaration of 'FloatDenormStyle'
```
</details>

Unfortunately, I couldn't find a good workaround for this. I think we need to mark this feature as blocked until the compiler bug is resolved.

Reply:
The compiler bug has been fixed for 17.6 Preview 3.